components/led_strip/led_strip.c

```diff
@@ -106,30 +106,50 @@ static void led_strip_fill_rmt_items(struct led_color_t *led_strip_buf, rmt_item
 static void led_strip_task(void *arg)
 {
     struct led_strip_t *led_strip = (struct led_strip_t *)arg;
-    portMUX_TYPE mux;
-    vPortCPUInitializeMutex(&mux);
 
     size_t num_items_malloc = (LED_STRIP_NUM_RMT_ITEMS_PER_LED * led_strip->led_strip_length);
     rmt_item32_t *rmt_items = (rmt_item32_t*) malloc(sizeof(rmt_item32_t) * num_items_malloc);
 
+    bool make_new_rmt_items = true;
+    bool prev_showing_buf_1 = !led_strip->showing_buf_1;
+
     if (!rmt_items) {
         vTaskDelete(NULL);
     }
 
-    for(;;) {
-        taskENTER_CRITICAL(&mux);
+    portMUX_TYPE mux;
```

Comment:
makes me nervous, move initialization elsewhere to ensure its init by the time led_strip_show is called